plugins/permission-backend/src/service/router.ts

```diff
@@ -97,15 +99,72 @@ export interface RouterOptions {
   policy: PermissionPolicy;
   identity: IdentityApi;
   config: Config;
+  cache?: CacheService;
 }
 
-const handleRequest = async (
-  requests: IdentifiedPermissionMessage<EvaluatePermissionRequest>[],
-  user: BackstageIdentityResponse | undefined,
-  policy: PermissionPolicy,
-  permissionIntegrationClient: PermissionIntegrationClient,
-  authHeader?: string,
-): Promise<IdentifiedPermissionMessage<EvaluatePermissionResponse>[]> => {
+const convertDecision = (options: {
+  id: string;
+  request: EvaluatePermissionRequest;
+  decision: PolicyDecision;
+  applyConditionsLoaderFor: Function;
+  resourceRef?: string;
+}): IdentifiedPermissionMessage<EvaluatePermissionResponse> => {
+  const { id, request, decision, applyConditionsLoaderFor, resourceRef } =
+    options;
+  if (decision.result !== AuthorizeResult.CONDITIONAL) {
+    return {
+      id,
+      ...decision,
+    };
+  }
+
+  if (!isResourcePermission(request.permission)) {
+    throw new Error(
+      `Conditional decision returned from permission policy for non-resource permission ${request.permission.name}`,
+    );
+  }
+
+  if (decision.resourceType !== request.permission.resourceType) {
+    throw new Error(
+      `Invalid resource conditions returned from permission policy for permission ${request.permission.name}`,
+    );
+  }
+
+  if (!resourceRef) {
+    return {
+      id,
+      ...decision,
+    };
+  }
+
+  return applyConditionsLoaderFor(decision.pluginId).load({
+    id,
+    resourceRef,
+    ...decision,
+  });
+};
+
+const handleRequest = async (options: {
+  requests: IdentifiedPermissionMessage<EvaluatePermissionRequest>[];
+  user: BackstageIdentityResponse | undefined;
+  policy: PermissionPolicy;
+  permissionIntegrationClient: PermissionIntegrationClient;
+  cacheEnabled?: boolean;
+  cacheTtl: number;
+  cache?: CacheService;
+  authHeader?: string;
+}): Promise<IdentifiedPermissionMessage<EvaluatePermissionResponse>[]> => {
+  const {
+    requests,
+    user,
+    policy,
+    permissionIntegrationClient,
+    cacheEnabled,
+    cacheTtl,
+    cache,
```

Comment:
code below will be easier to write if this is just a single

```ts
cache?: {
  ttl: number;
  service: CacheService;
},
```

and then code can just do `if (cache)` because that signals that it's enabled, and then both those fields WILL be present (there's no enabled flag needed)